### ISSUE TITLE
Investigate doc building with workflow

### ISSUE THREAD

**Issue by flying-sheep** [scverse/cookiecutter-scverse#314]
If RTD was a GitHub app, it would be super simple, but the way it is, maybe setting up [sphinx-action](https://github.com/ammaraskar/sphinx-action) + GitHub pages is easier? We could check.

**Comment by grst** [comment]
Last time I used such an approach, there were two downsides:
* it's hard to keep multiple versions
* github actions dont run on PRs from forks

But maybe there are solutions for this by now

**Comment by flying-sheep** [comment]
Yeah, I was inspired by [the workflow that came with the Markdown presentation tool Marp](https://github.com/scverse/cookiecutter-scverse-presentation/blob/2b86c40bd1fa7bdd832a254c6c5a195334c6b612/.github/workflows/marp-to-pages.yml), because of its simple PR preview support. It works like this:

1. configure GH pages the old way (deploy by pushing to `gh-pages` branch)
2. build docs
3. In PRs, [`rossjrw/pr-preview-action`](https://github.com/rossjrw/pr-preview-action) updates a directory in the `gh-pages` branch called `pr-previews`
4. On the `main` branch, things are just deployed to the root of the `gh-pages` branch, ignoring the `pr-previews` directory

Future changes regarding forks:

- The fork problem can apparently be solved and will be in the preview action: https://github.com/rossjrw/pr-preview-action/pull/6
- GitHub is working on its own PR preview mechanism for GH pages

Due to the security issues of the first approach, waiting for the second one is probably best.

**Comment by grst** [comment]
The native github preview mechanism sounds promising. Do you also have a solution for keeping different versions (and linking to them from the documentation)? 
Afaik readthedocs somehow injects the version picker into the docs website after it is already built, which allows to pick newer versions from older docs. 

I used to have something like that in scirpy before I moved to RTD, using some javascript hacks, but it wasn't the most elegant solution: https://github.com/scverse/scirpy/blob/6ccaa6673d3d043e1e7988e5c92c037ad2ffc713/docs/versions.rst?plain=1#L13-L21
